src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -51,38 +51,30 @@ internal NpgsqlBinaryImporter(NpgsqlConnector connector, string copyFromCommand)
             _buf = connector.WriteBuffer;
             _column = -1;
 
-            try
-            {
-                _connector.SendQuery(copyFromCommand);
+            _connector.SendQuery(copyFromCommand);
 
-                CopyInResponseMessage copyInResponse;
-                var msg = _connector.ReadMessage();
-                switch (msg.Code)
-                {
-                case BackendMessageCode.CopyInResponse:
-                    copyInResponse = (CopyInResponseMessage)msg;
-                    if (!copyInResponse.IsBinary)
-                        throw new ArgumentException("copyFromCommand triggered a text transfer, only binary is allowed", nameof(copyFromCommand));
-                    break;
-                case BackendMessageCode.CompletedResponse:
-                    throw new InvalidOperationException(
-                        "This API only supports import/export from the client, i.e. COPY commands containing TO/FROM STDIN. " +
-                        "To import/export with files on your PostgreSQL machine, simply execute the command with ExecuteNonQuery. " +
-                        "Note that your data has been successfully imported/exported.");
-                default:
-                    throw _connector.UnexpectedMessageReceived(msg.Code);
-                }
-
-                NumColumns = copyInResponse.NumColumns;
-                _params = new NpgsqlParameter[NumColumns];
-                _buf.StartCopyMode();
-                WriteHeader();
-            }
-            catch
+            CopyInResponseMessage copyInResponse;
+            var msg = _connector.ReadMessage();
+            switch (msg.Code)
             {
-                _connector.Break();
-                throw;
+            case BackendMessageCode.CopyInResponse:
+                copyInResponse = (CopyInResponseMessage)msg;
+                if (!copyInResponse.IsBinary)
```

Comment:
Note that the connection will still be broken for this error, and also for CompletedResponse below. That's totally fine, but indeed harder to take care of cleanly - just mentioning it.

Reply:
Thanks for review.
I investigated behavior with BackendMessageCode.CopyInResponse case when table exists but TEXT copy syntax was used (COPY temp_table (blob) FROM STDIN).
Although connector state is not changed subsequent queries will fail with unexpected errors, eg:
"SELECT 1" will fail with PostgresException : 08P01: unexpected message type 0x50 during COPY from stdin.
Seems in that case we should either break the connector or try to make it work if it's possible.

```
using (var conn = OpenConnection())
{
    conn.ExecuteNonQuery("create temp table temp_table (blob bytea)");

    // using text copy syntax within BeginBinaryImport
    Assert.Throws<ArgumentException>(() => conn.BeginBinaryImport("COPY temp_table (blob) FROM STDIN"));

    // connection seems to be open
    Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Open));

    // PostgresException : 08P01: unexpected message type 0x50 during COPY from stdin.
    Assert.That(conn.ExecuteScalar("SELECT 1"), Is.EqualTo(1));
}
```

Reply:
It's possible to "cancel" import if it's not binary by sending CopyFailMessage, like NpgsqlRawCopyStream.Cancel does.
After that subsequent queries work as expected.

Reply:
I'm fine with breaking the connection in this kind of case, as it's a very clear programmer error - I don't think that the effort to recover the connection is worth it in this case.